lib/libzfs/libzfs_import.c

```diff
@@ -820,12 +820,31 @@ pool_active(libzfs_handle_t *hdl, const char *name, uint64_t guid,
 	return (0);
 }
 
+boolean_t
+zfs_forceimport_required(nvlist_t *config)
+{
+	uint64_t state;
+	uint64_t hostid = 0;
+	unsigned long system_hostid = get_system_hostid();
+
+	state = fnvlist_lookup_uint64(config, ZPOOL_CONFIG_POOL_STATE);
+	(void) nvlist_lookup_uint64(config, ZPOOL_CONFIG_HOSTID,
+	    &hostid);
```

Comment:
Hm, I'm wrong; zfs_force_import_required() does use hostid != 0.  But I do not think it should.  I also have an issue with the state comparison.  I describe both in a comment within this function so the concerns are not lost.